code/iaas/service-discovery/server/src/main/java/io/cattle/platform/servicediscovery/deployment/impl/DeploymentUnit.java

```diff
@@ -194,6 +194,10 @@ public boolean isComplete() {
     protected Map<String, String> getLabels(DeploymentUnitInstance instance) {
         Service service = instance.getService();
         Map<String, String> labels = new HashMap<>();
+
+        // TODO: This might change in the future if we merge the labels from all services
+        labels.putAll(context.sdService.getServiceLabels(service));
```

Comment:
No need to populate the labels here. Service labels get set on the instance in ServiceDiscoveryServiceImpl.buildServiceInstanceLaunchData call, along with other parameters defined on the service level. In this code only deployment unit related labels get populated.